engine/src/org/pentaho/di/trans/steps/sort/SortRowsMeta.java

```diff
@@ -320,7 +341,8 @@ public void getFields( RowMetaInterface inputRowMeta, String name, RowMetaInterf
         ValueMetaInterface valueMeta = inputRowMeta.getValueMeta( idx );
         valueMeta.setSortedDescending( !ascending[i] );
         valueMeta.setCaseInsensitive( !caseSensitive[i] );
-
+        valueMeta.setCollatorDisabled( !collatorEnabled[i] );
```

Comment:
I understand, and I see that the pattern is as you've described. I hate it, but perhaps it's best to leave it as you have it.